pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/main/java/org/apache/pinot/plugin/stream/kafka20/KafkaPartitionLevelConnectionHandler.java

```diff
@@ -56,6 +56,8 @@ public abstract class KafkaPartitionLevelConnectionHandler {
   protected final Consumer<String, Bytes> _consumer;
   protected final TopicPartition _topicPartition;
   protected final Properties _consumerProp;
+  // Admin client for thread-safe offset lookups
+  protected final AdminClient _adminClient;
```

Comment:
What is the overhead of creating the `AdminClient`? Should we create both `Consumer` and `AdminClient` just when needed, and close immediately after use?

Reply:
The consumer creation is definitely slow as it tries to fetch metadata for the topic on creation. 

Let me check for Admin client

Reply:
Admin client overhead is less than 100ms always
Consumer client overhead is an order of magnitude more sometimes

Change the PR so that we create admin client on demand